url294e6c45f3_Ecdosis_NMergeNew_tgz-pJ8-TestJ8/src/edu/luc/nmerge/graph/SimpleQueue.java

```diff
@@ -0,0 +1,112 @@
+/*
+ *  NMerge is Copyright 2009 Desmond Schmidt
+ * 
+ *  This file is part of NMerge. NMerge is a Java library for merging 
+ *  multiple versions into multi-version documents (MVDs), and for 
+ *  reading, searching and comparing them.
+ *
+ *  NMerge is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NMerge is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package edu.luc.nmerge.graph;
+
+/**
+ * A simple queue implementation designed to remove suspicious 
+ * interactions with LinkedList during debugging. Perhaps could be 
+ * replaced by LinkedList now.
+ * @author Desmond Schmidt
+ */
+public class SimpleQueue<E> 
+{
+	Item head,tail;
+	int size;
+	/**
+	 * This is just an item in the queue represented as a list
+	 */
+	private class Item
+	{
+		Item next;
+		E item;
+		Item( E item )
+		{
+			this.item = item;
+		}
+	}
+    /**
+     * Retrieves and removes the head of this queue, or null if this 
+     * queue is empty.
+     * @return the head of the queue
+     */
+	public E poll()
+	{
+		if ( head == null )
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of == on line 52 to != check before setting the field value on line 57 or verify if the check is needed at all.